### PR TITLE
*: bump some dependencies for release-6.5 hot patch

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve?

Issue Number: NA

Problem Summary:

### What changed and how does it work?

The change is made for the following dependencies to prevent some potential security vulnerabilities:

* google.golang.org/grpc v1.51.0 -> v1.59.0
* github.com/lestrrat-go/jwx/v2 v2.0.17 -> v2.0.21
* golang.org/x/crypto v0.15.0 -> v0.24.0
* golang.org/x/net v0.18.0 -> v0.26.0
* google.golang.org/protobuf v1.28.1 -> v1.33.0

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- NA

Documentation

- NA

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/testkit.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func RunTestMain(m *testing.M) {
 	})
 	tikv.EnableFailpoints()
 	opts := []goleak.Option{
-		goleak.IgnoreTopFunction("github.com/golang/glog.(*loggingT).flushDaemon"),
+		goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon"),
```
Comment (YangKeao) [inline tests/realtikvtest/testkit.go:61]: Maybe we should also ignore `google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer` leak in `realtikvtest`.

ref https://github.com/pingcap/tidb/pull/51496

Comment (hawkingrei) [reply tests/realtikvtest/testkit.go:61]: it has been updated.

**Review by hawkingrei — APPROVED** [review]

**Review by bb7133 — COMMENTED** [review]
Need another reviewer

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**